{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# Your first basic pipeline\n",
    "\n",
    "## The basic functions\n",
    "\n",
    "Pypiper is simple but powerful. Your pipeline is a python script, let's call it `pipeline.py`. You really only need to know about 3 commands to get started:\n",
    "\n",
    "* `PipelineManager()`: PipelineManager constructor, initializes the pipeline.\n",
    "* `PipelineManager.run()`: The primary workhorse function; runs a command.\n",
    "* `PipelineManager.stop_pipeline()`: Terminate the pipeline.\n",
    "\n",
    "That's all you need to create a powerful pipeline. You can find in-depth reference documentation for each method in the API. In particular, most of Pypiperâ€™s power comes from the `run` method, which has a series of options outlined in [dedicated documentation on the run method](/advanced-run-method).\n",
    "\n",
    "To write your first basic pipeline, first `import pypiper`, then specify an output folder and create a new `PipelineManager` object:\n",
    "\n",
    "\t#!/usr/bin/env python\n",
    "\timport pypiper, os\n",
    "\toutfolder = \"pipeline_output/\"  # Choose a folder for your results\n",
    "\tpm = pypiper.PipelineManager(name=\"my_pipeline\", outfolder=outfolder)\n",
    "\n",
    "Creating the `PipelineManger` object creates your `outfolder` and places a flag called `my_pipeline_running.flag` in the folder. It also initializes the log file (``my_pipeline_log.md``) with statistics such as time of starting, compute node, software versions, command-line parameters, etc.\n",
    "\n",
    "Now, the workhorse of `PipelineManager` is the `run()` function. Essentially, you create a shell command as a string in python, and then pass it and its target (a file it creates) to `run()`. The target is the final output file created by your command. Let's use the built-in `shuf` command to create some random numbers and put them in a file called `outfile.txt`:\n",
    "\n",
    "\t# our command will produce this output file\n",
    "\ttarget = os.path.join(outfolder, \"outfile.txt\")\n",
    "\tcommand = \"shuf -i 1-500000000 -n 10000000 > \" + target\n",
    "\tpm.run(command, target)\n",
    "\n",
    "\n",
    "The command (`command`) is the only required argument to `run()`. You can leave `target` empty (pass `None`). If you **do** specify a target, the command will only be run if the target file does not already exist. If you **do not** specify a target, the command will be run every time the pipeline is run. \n",
    "\n",
    "Now string together whatever commands your pipeline requires! At the end, terminate the pipeline so it gets flagged as successfully completed:\n",
    "\n",
    "\tpm.stop_pipeline()\n",
    "\n",
    "That's it! By running commands through `run()` instead of directly in bash, you get a robust, logged, restartable pipeline manager for free!\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "\n",
    "## Advanced functions\n",
    "Once you've mastered the basics, add in a few other commands that make debugging and development easier:\n",
    "\n",
    "* `timestamp()`: Print message, time, and time elapsed, perhaps creating checkpoint.\n",
    "* `clean_add()`: Add files (or regexs) to a cleanup list, to delete when this pipeline completes successfully.\n",
    "\n",
    "## Reporting results \n",
    "`Pypiper` also has a really nice system for collecting and aggregating result statistics and pipeline outputs. \n",
    "\n",
    "* `report_result()`: Writes a line in the `stats.tsv` file with a key-value pair of some pipeline result.\n",
    "* `report_object()`: Writes a line in the `objects.tsv` file with a key-value pair pointing to any file (such as an image, HTML report, or other) created by the pipeline.\n",
    "* `get_stat()`:\tReturns a stat that was previously reported.\n",
    "\n",
    "## Putting them together in `basic.py`\n",
    "Now, download [basic.py](https://github.com/databio/pypiper/blob/master/example_pipelines/basic.py) and run it with `python basic.py` (or, better yet, make it executable (`chmod 755 basic.py`) and then run it directly with `./basic.py`). This example is a documented vignette; so just read it and run it to get an idea of how things work. Here are the contents of `basic.py`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# %load ../example_pipelines/basic.py\n",
    "#!/usr/bin/env python\n",
    "\n",
    "\"\"\"Getting Started: A simple sample pipeline built using pypiper.\"\"\"\n",
    "\n",
    "# This is a runnable example. You can run it to see what the output\n",
    "# looks like.\n",
    "\n",
    "# First, make sure you can import the pypiper package\n",
    "\n",
    "import os\n",
    "import pypiper\n",
    "\n",
    "# Create a PipelineManager instance (don't forget to name it!)\n",
    "# This starts the pipeline.\n",
    "\n",
    "pm = pypiper.PipelineManager(name=\"BASIC\",\n",
    "    outfolder=\"pipeline_output/\")\n",
    "\n",
    "# Now just build shell command strings, and use the run function\n",
    "# to execute them in order. run needs 2 things: a command, and the\n",
    "# target file you are creating.\n",
    "\n",
    "# First, generate some random data\n",
    "\n",
    "# specify target file:\n",
    "tgt = \"pipeline_output/test.out\"\n",
    "\n",
    "# build the command\n",
    "cmd = \"shuf -i 1-500000000 -n 10000000 > \" + tgt\n",
    "\n",
    "# and run with run().\n",
    "pm.run(cmd, target=tgt)\n",
    "\n",
    "# Now copy the data into a new file.\n",
    "# first specify target file and build command:\n",
    "tgt = \"pipeline_output/copied.out\"\n",
    "cmd = \"cp pipeline_output/test.out \" + tgt\n",
    "pm.run(cmd, target=tgt)\n",
    "\n",
    "# You can also string multiple commands together, which will execute\n",
    "# in order as a group to create the final target.\n",
    "cmd1 = \"sleep 5\"\n",
    "cmd2 = \"touch pipeline_output/touched.out\"\n",
    "pm.run([cmd1, cmd2], target=\"pipeline_output/touched.out\")\n",
    "\n",
    "# A command without a target will run every time.\n",
    "# Find the biggest line\n",
    "cmd = \"awk 'n < $0 {n=$0} END{print n}' pipeline_output/test.out\"\n",
    "pm.run(cmd, \"lock.max\")\n",
    "\n",
    "# Use checkprint() to get the results of a command, and then use\n",
    "# report_result() to print and log key-value pairs in the stats file:\n",
    "last_entry = pm.checkprint(\"tail -n 1 pipeline_output/copied.out\")\n",
    "pm.report_result(\"last_entry\", last_entry)\n",
    "\n",
    "\n",
    "# Now, stop the pipeline to complete gracefully.\n",
    "pm.stop_pipeline()\n",
    "\n",
    "# Observe your outputs in the pipeline_output folder \n",
    "# to see what you've created.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# Building a pipeline to count number of reads\n",
    "\n",
    "Here we have a more advanced bioinformatics pipeline that adds some new concepts. This is a simple script that takes an input file and returns the file size and the number of sequencing reads in that file. This example uses a function from from the built-in :doc:`NGSTk toolkit <ngstk>`. In particular, this toolkit contains a few handy functions that make it easy for a pipeline to accept inputs of various types. So, this pipeline can count the number of reads from files in ``BAM`` format, or ``fastq`` format, or ``fastq.gz`` format. You can also use the same functions from NGSTk to develop a pipeline to do more complicated things, and handle input of any of these types.\n",
    "\n",
    "First, grab this pipeline. Download [count_reads.py](https://github.com/databio/pypiper/blob/master/example_pipelines/count_reads.py), make it executable (`chmod 755 count_reads.py`), and then run it with `./count_reads.py`). \n",
    "\n",
    "You can grab a few small data files in the [microtest repository](https://github.com/epigen/microtest/tree/master/config). Run a few of these files like this:\n",
    "\n",
    "```\n",
    "./count_reads.py -I ~/code/microtest/data/rrbs_PE_R1.fastq.gz -O $HOME -S sample1\n",
    "./count_reads.py -I ~/code/microtest/data/rrbs_PE_fq_R1.fastq -O $HOME -S sample2\n",
    "./count_reads.py -I ~/code/microtest/data/atac-seq_SE.bam -O $HOME -S sample3\n",
    "```\n",
    "\n",
    "This example is a documented vignette; so just read it and run it to get an idea of how things work.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# %load ../example_pipelines/count_reads.py\n",
    "#!/usr/bin/env python\n",
    "\n",
    "\"\"\"\n",
    "Counts reads.\n",
    "\"\"\"\n",
    "\n",
    "__author__ = \"Nathan Sheffield\"\n",
    "__email__ = \"nathan@code.databio.org\"\n",
    "__license__ = \"GPL3\"\n",
    "__version__ = \"0.1\"\n",
    "\n",
    "from argparse import ArgumentParser\n",
    "import os, re\n",
    "import sys\n",
    "import subprocess\n",
    "import yaml\n",
    "import pypiper\n",
    "\n",
    "parser = ArgumentParser(\n",
    "    description=\"A pipeline to count the number of reads and file size. Accepts\"\n",
    "    \" BAM, fastq, or fastq.gz files.\")\n",
    "\n",
    "# First, add standard arguments from Pypiper.\n",
    "# groups=\"pypiper\" will add all the arguments that pypiper uses,\n",
    "# and adding \"common\" adds arguments for --input and --sample--name\n",
    "# and \"output_parent\". You can read more about your options for standard\n",
    "# arguments in the pypiper docs (section \"command-line arguments\")\n",
    "parser = pypiper.add_pypiper_args(parser, groups=[\"pypiper\", \"common\", \"ngs\"],\n",
    "                                    args=[\"output-parent\", \"config\"],\n",
    "                                    required=['sample-name', 'output-parent'])\n",
    "\n",
    "# Add any pipeline-specific arguments if you like here.\n",
    "\n",
    "args = parser.parse_args()\n",
    "\n",
    "if not args.input or not args.output_parent:\n",
    "    parser.print_help()\n",
    "    raise SystemExit\n",
    "\n",
    "if args.single_or_paired == \"paired\":\n",
    "    args.paired_end = True\n",
    "else:\n",
    "    args.paired_end = False\n",
    "\n",
    "# args for `output_parent` and `sample_name` were added by the standard \n",
    "# `add_pypiper_args` function. \n",
    "# A good practice is to make an output folder for each sample, housed under\n",
    "# the parent output folder, like this:\n",
    "outfolder = os.path.abspath(os.path.join(args.output_parent, args.sample_name))\n",
    "\n",
    "# Create a PipelineManager object and start the pipeline\n",
    "pm = pypiper.PipelineManager(name=\"count\",\n",
    "                             outfolder=outfolder, \n",
    "                             args=args)\n",
    "\n",
    "# NGSTk is a \"toolkit\" that comes with pypiper, providing some functions\n",
    "# for dealing with genome sequence data. You can read more about toolkits in the\n",
    "# documentation\n",
    "\n",
    "# Create a ngstk object\n",
    "ngstk = pypiper.NGSTk(pm=pm)\n",
    "\n",
    "raw_folder = os.path.join(outfolder, \"raw/\")\n",
    "fastq_folder = os.path.join(outfolder, \"fastq/\")\n",
    "\n",
    "# Merge/Link sample input and Fastq conversion\n",
    "# These commands merge (if multiple) or link (if single) input files,\n",
    "# then convert (if necessary, for bam, fastq, or gz format) files to fastq.\n",
    "\n",
    "# We'll start with a timestamp that will provide a division for this section\n",
    "# in the log file\n",
    "pm.timestamp(\"### Merge/link and fastq conversion: \")\n",
    "\n",
    "# Now we'll rely on 2 NGSTk functions that can handle inputs of various types\n",
    "# and convert these to fastq files.\n",
    "\n",
    "local_input_files = ngstk.merge_or_link(\n",
    "                        [args.input, args.input2],\n",
    "                        raw_folder,\n",
    "                        args.sample_name)\n",
    "\n",
    "cmd, out_fastq_pre, unaligned_fastq = ngstk.input_to_fastq(\n",
    "                                            local_input_files,\n",
    "                                            args.sample_name,\n",
    "                                            args.paired_end,\n",
    "                                            fastq_folder)\n",
    "\n",
    "\n",
    "# Now we'll use another NGSTk function to grab the file size from the input files\n",
    "#\n",
    "pm.report_result(\"File_mb\", ngstk.get_file_size(local_input_files))\n",
    "\n",
    "\n",
    "# And then count the number of reads in the file\n",
    "\n",
    "n_input_files = len(filter(bool, local_input_files))\n",
    "\n",
    "raw_reads = sum([int(ngstk.count_reads(input_file, args.paired_end)) \n",
    "                for input_file in local_input_files]) / n_input_files\n",
    "\n",
    "# Finally, we use the report_result() function to print the output and \n",
    "# log the key-value pair in the standard stats.tsv file\n",
    "pm.report_result(\"Raw_reads\", str(raw_reads))\n",
    "\n",
    "# Cleanup\n",
    "pm.stop_pipeline()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
