# PEP in practice

## Software using PEP

Publicly available software that builds on PEP:

* [PEPATAC](http://pepatac.databio.org/) - An ATAC-seq pipeline. 
* [PEPPRO](http://peppro.databio.org/) - An nascent RNA profiling pipeline (PRO-seq, GRO-seq, ChRO-seq).
* [peppy](https://github.com/pepkit/peppy)
* [pepr](https://github.com/pepkit/pepr)
* [geofetch](https://github.com/pepkit/geofetch) - Converts GEO or SRA accessions into PEP projects.
* [divcfg](https://github.com/pepkit/divcfg)
* [pifaces](https://github.com/pepkit/pifaces)
* [pypiper](https://github.com/databio/pypiper)
* [dnameth_pipelines](https://github.com/databio/dnameth_pipelines)
* [projectInit](https://github.com/databio/projectInit)
* [ngstoolkit](https://github.com/afrendeiro/toolkit) - NGS analysis toolkit
* [BiocProject](https://github.com/pepkit/BiocProject)

## Demo data using PEP

* [example_peps repository](https://github.com/pepkit/example_peps) - A collection of example PEPs demonstrating various features.
* [microtest](https://github.com/epigen/microtest)
* [hello looper! example](https://github.com/pepkit/hello_looper)

## Real datasets organized in PEP format:

* [https://github.com/epigen/crop-seq](https://github.com/epigen/crop-seq)
* [https://github.com/epigen/baf_complex](https://github.com/epigen/baf_complex)
* [https://github.com/epigen/mthfd1](https://github.com/epigen/mthfd1)
* [https://github.com/epigen/cll-ibrutinib_time](https://github.com/epigen/cll-ibrutinib_time)
* [https://github.com/epigen/cll-ibrutinib](https://github.com/epigen/cll-ibrutinib)
* [https://github.com/epigen/cll-chromatin](https://github.com/epigen/cll-chromatin)

## Publications that use PEP software:

<ul>
<li><b>Nash et al. (2023). </b><i>Maternal diet alters long-term innate immune cell memory in fetal and juvenile hematopoietic stem and progenitor cells in nonhuman primate offspring</i> 
<br><i>Cell Reports</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1016/j.celrep.2023.112393">10.1016/j.celrep.2023.112393</a></span></li>
<li><b>Danko et al. (2023). </b><i>Evolution of promoter-proximal pausing enabled a new layer of transcription control</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.21203/rs.3.rs-2679520/v1">10.21203/rs.3.rs-2679520/v1</a></span></li>
<li><b>Singh et al. (2023). </b><i>Cohesin regulates alternative splicing</i> 
<br><i>Science Advances</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1126/sciadv.ade3876">10.1126/sciadv.ade3876</a></span></li>
<li><b>Luo et al. (2023). </b><i>Epiblast-like stem cells established by Wnt/-catenin signaling manifest distinct features of formative pluripotency and germline competence</i> 
<br><i>Cell Reports</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1016/j.celrep.2023.112021">10.1016/j.celrep.2023.112021</a></span></li>
<li><b>Abadie et al. (2022). </b><i>Flexible and scalable control of T cell memory by a reversible epigenetic switch</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.1101/2022.12.31.521782">10.1101/2022.12.31.521782</a></span></li>
<li><b>Wolpe et al. (2022). </b><i>Correction of transposase sequence bias in ATAC-seq data with rule ensemble modeling</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.1101/2022.12.08.519600">10.1101/2022.12.08.519600</a></span></li>
<li><b>Robbe et al. (2022). </b><i>Whole-genome sequencing of chronic lymphocytic leukemia identifies subgroups with distinct biological and clinical features</i> 
<br><i>Nature Genetics</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41588-022-01211-y">10.1038/s41588-022-01211-y</a></span></li>
<li><b>Robey et al. (2022). </b><i>The methyltransferases METTL7A and METTL7B confer resistance to thiol-based histone deacetylase inhibitors</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.1101/2022.10.07.511310">10.1101/2022.10.07.511310</a></span></li>
<li><b>Callahan et al. (2022). </b><i>High enhancer activity is an epigenetic feature of HPV negative atypical head and neck squamous cell carcinoma</i> 
<br><i>Frontiers in Cell and Developmental Biology</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.3389/fcell.2022.936168">10.3389/fcell.2022.936168</a></span></li>
<li><b>Duvall et al. (2022). </b><i>Single-cell transcriptome and accessible chromatin dynamics during endocrine pancreas development</i> 
<br><i>Proceedings of the National Academy of Sciences</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1073/pnas.2201267119">10.1073/pnas.2201267119</a></span></li>
<li><b>Taklifi et al. (2022). </b><i>Integrating chromatin accessibility states in the design of targeted sequencing panels for liquid biopsy</i> 
<br><i>Scientific Reports</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41598-022-14675-z">10.1038/s41598-022-14675-z</a></span></li>
<li><b>Grandi et al. (2022). </b><i>Chromatin accessibility profiling by ATAC-seq</i> 
<br><i>Nature Protocols</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41596-022-00692-9">10.1038/s41596-022-00692-9</a></span></li>
<li><b>Hunter et al. (2022). </b><i>HNF4A modulates glucocorticoid action in the liver</i> 
<br><i>Cell Reports</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1016/j.celrep.2022.110697">10.1016/j.celrep.2022.110697</a></span></li>
<li><b>O'Connor et al. (2022). </b><i>BET Protein Inhibition Regulates Macrophage Chromatin Accessibility and Microbiota-Dependent Colitis</i> 
<br><i>Frontiers in Immunology</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.3389/fimmu.2022.856966">10.3389/fimmu.2022.856966</a></span></li>
<li><b>Wang et al. (2022). </b><i>Prediction of histone post-translational modification patterns based on nascent transcription data</i> 
<br><i>Nature Genetics</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41588-022-01026-x">10.1038/s41588-022-01026-x</a></span></li>
<li><b>Zhang et al. (2022). </b><i>Extensive evaluation of ATAC-seq protocols for native or formaldehyde-fixed nuclei</i> 
<br><i>BMC Genomics</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1186/s12864-021-08266-x">10.1186/s12864-021-08266-x</a></span></li>
<li><b>Shahin et al. (2021). </b><i>Germline biallelic mutation affecting the transcription factor Helios causes pleiotropic defects of immunity</i> 
<br><i>Science Immunology</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1126/sciimmunol.abe3981">10.1126/sciimmunol.abe3981</a></span></li>
<li><b>Ram-Mohan et al. (2021). </b><i>Profiling chromatin accessibility responses in human neutrophils with sensitive pathogen detection</i> 
<br><i>Life Science Alliance</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.26508/lsa.202000976">10.26508/lsa.202000976</a></span></li>
<li><b>Robertson et al. (2021). </b><i>Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes</i> 
<br><i>Nature Genetics</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41588-021-00880-5">10.1038/s41588-021-00880-5</a></span></li>
<li><b>Cheung et al. (2021). </b><i>Repression of CTSG, ELANE and PRTN3-mediated histone H3 proteolytic cleavage promotes monocyte-to-macrophage differentiation</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41590-021-00928-y">10.1038/s41590-021-00928-y</a></span></li>
<li><b>Hasegawa et al. (2021). </b><i>Clonal inactivation of telomerase promotes accelerated stem cell differentiation</i> 
<br><i>bioRxiv</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1101/2021.04.28.441728">10.1101/2021.04.28.441728</a></span></li>
<li><b>Weber et al. (2021). </b><i>Transient rest restores functionality in exhausted CAR-T cells through epigenetic remodeling</i> 
<br><i>Science</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1126/science.aba1786">10.1126/science.aba1786</a></span></li>
<li><b>Gharavi et al. (2021). </b><i>Embeddings of genomic region sets capture rich biological associations in low dimensions</i> 
<br><i>Bioinformatics</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1093/bioinformatics/btab439">10.1093/bioinformatics/btab439</a></span></li>
<li><b>Tovar et al. (2021). </b><i>Integrative phenotypic and genomic analyses reveal strain-dependent responses to acute ozone exposure and their associations with airway macrophage transcriptional activity</i> 
<br><i>bioRxiv</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1101/2021.01.29.428733">10.1101/2021.01.29.428733</a></span></li>
<li><b>Granja et al. (2021). </b><i>ArchR is a scalable software package for integrative single-cell chromatin accessibility analysis</i> 
<br><i>Nature Genetics</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41588-021-00790-6">10.1038/s41588-021-00790-6</a></span></li>
<li><b>Gu et al. (2021). </b><i>Bedshift: perturbation of genomic interval sets</i> 
<br><i>Genome Biology</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1186/s13059-021-02440-w">10.1186/s13059-021-02440-w</a></span></li>
<li><b>MÃ¶lder et al. (2021). </b><i>Sustainable data analysis with Snakemake</i> 
<br><i>F1000Research</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.12688/f1000research.29032.2">10.12688/f1000research.29032.2</a></span></li>
<li><b>Smith et al. (2021). </b><i>PEPPRO: quality control and processing of nascent RNA profiling data</i> 
<br><i>Genome Biology</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1186/s13059-021-02349-4">10.1186/s13059-021-02349-4</a></span></li>
<li><b>Fan et al. (2020). </b><i>Epigenomic Reprogramming toward Mesenchymal-Epithelial Transition in Ovarian-Cancer-Associated Mesenchymal Stem Cells Drives Metastasis</i> 
<br><i>Cell Reports</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1016/j.celrep.2020.108473">10.1016/j.celrep.2020.108473</a></span></li>
<li><b>ROBERTSON et al. (2020). </b><i>112-OR: Integrative Analysis of Chromatin Accessibility and Genetic Risk in T1D Patients and Controls</i> 
<br><i>Diabetes</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.2337/db20-112-or">10.2337/db20-112-or</a></span></li>
<li><b>Smith and Sheffield (2020). </b><i>Analytical Approaches for ATAC-seq Data Analysis</i> 
<br><i>Current Protocols in Human Genetics</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1002/cphg.101">10.1002/cphg.101</a></span></li>
<li><b>Liu (2020). </b><i>Clinical implications of chromatin accessibility in human cancers</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.18632/oncotarget.27584">10.18632/oncotarget.27584</a></span></li>
<li><b>Zhou et al. (2020). </b><i>CATA: a comprehensive chromatin accessibility database for cancer</i> 
<br><i>bioRxiv</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1101/2020.05.16.099325">10.1101/2020.05.16.099325</a></span></li>
<li><b>Wang et al. (2020). </b><i>Interdependence between histone marks and steps in Pol II transcription</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.1101/2020.04.08.032730">10.1101/2020.04.08.032730</a></span></li>
<li><b>Cai et al. (2020). </b><i>Specific chromatin landscapes and transcription factors couple breast cancer subtype with metastatic relapse to lung or brain</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.1186/s12920-020-0695-0">10.1186/s12920-020-0695-0</a></span></li>
<li><b>Li et al. (2020). </b><i>Acetate supplementation restores chromatin accessibility and promotes tumor cell differentiation under hypoxia</i> 
<br><i>Cell Death &amp$$ Disease</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/s41419-020-2303-9">10.1038/s41419-020-2303-9</a></span></li>
<li><b>Stolarczyk et al. (2020). </b><i>Refgenie: a reference genome resource manager</i> 
<br><i>GigaScience</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1093/gigascience/giz149">10.1093/gigascience/giz149</a></span></li>
<li><b>Liang et al. (2019). </b><i>Global changes in chromatin accessibility and transcription following ATRX inactivation in human cancer cells</i> 
<br> <span class="doi">DOI: <a href="http://dx.doi.org/10.1002/1873-3468.13549">10.1002/1873-3468.13549</a></span></li>
<li><b>Corces et al. (2018). </b><i>The chromatin accessibility landscape of primary human cancers</i> 
<br><i>Science</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1126/science.aav1898">10.1126/science.aav1898</a></span></li>
<li><b>Datlinger et al. (2017). </b><i>Pooled CRISPR screening with single-cell transcriptome readout</i> 
<br><i>Nat. Methods</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/nmeth.4177">10.1038/nmeth.4177</a></span></li>
<li><b>Sheffield et al. (2017). </b><i>DNA methylation heterogeneity defines a disease spectrum in Ewing sarcoma</i> 
<br><i>Nature Medicine</i>.  <span class="doi">DOI: <a href="http://dx.doi.org/10.1038/nm.4273">10.1038/nm.4273</a></span></li>

</ul>

### PEP shield

If your project is PEP-compatible, please add it to this list with a [pull request](https://github.com/pepkit/pepkit.github.io/blob/master/_docs/tools.md) and use this shield to showcase PEP:

<img src="http://pepkit.github.io/img/PEP-compatible-green.svg" alt="PEP compatible" style="float:left; margin:10px"><br clear="all"/>

Here's `markdown` (for use on GitHub READMEs):
```
[![PEP compatible](http://pepkit.github.io/img/PEP-compatible-green.svg)](http://pepkit.github.io)
```

Or `HTML`:
```
<a href="http://pepkit.github.io"><img src="http://pepkit.github.io/img/PEP-compatible-green.svg" alt="PEP compatible" style="float:left; margin:10px"></a>
```

